### DISCUSSION TITLE
Playlist persist on closing?

### DISCUSSION THREAD

**Discussion by illettante** [mywave82/opencubicplayer#119] [Q&A]
Hi folks, is it possible for OCP to store the playlist for when it's opened next time? This is admittedly an edge case situation - I have an offline collection of near 300,000 mods from mirroring archives, and i like to listen to the whole lot on shuffle so i never hear the same song twice, but it takes six or seven minutes to add them all every time i start ocp. I can't put them all in a playlist because OCP outputs that the pls file is too big. Fully understand this isn't a common use case, so if the answer is just a plain no, that's ok.

**Comment by mywave82** [comment]
Putting items on the playlist will cause a lot of internal structures to be created and memory allocated per file-entry.

Long term solution will probably be to verify how slow/fast medialib: is, and make it possible for the playlist to take different actions when it runs out of items, like to play random songs from medialib, the same directory, etc.

This will anyhow take some time. I am currently working on modland: support (currently hosted in the badly named tempmaster branch) and time is a bit sparse at the moment, since I have a toddler at home.

**Reply by mywave82** [reply]
How much memory does OCP consume for you when you have this many items in it?

**Reply by illettante** [reply]
I wasn't sure of the best way to measure this for you - i ran `ocp-curses`, added the whole directory of files and have been polling it about every 5 minutes with `ps`. It started using about 92kb ram and climbed to about 107kb over half an hour of playtime, and it's been sitting at about there for another half an hour, give or take 1kb.

I understand RE: toddlers being time-intensive - I have a child turning three shortly! This is a FOSS project, you're working on it out of the goodness of your heart and you're under no obligation to pursue any users' whims. I'm grateful for the time you've spent on it already.
